features/steps/permissions_component.go

```diff
@@ -0,0 +1,154 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	componenttest "github.com/ONSdigital/dp-component-test"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	dpMongoDriver "github.com/ONSdigital/dp-mongodb/v2/mongodb"
+	"github.com/ONSdigital/dp-permissions-api/api"
+	"github.com/ONSdigital/dp-permissions-api/config"
+	"github.com/ONSdigital/dp-permissions-api/mongo"
+	"github.com/ONSdigital/dp-permissions-api/service"
+	serviceMock "github.com/ONSdigital/dp-permissions-api/service/mock"
+	"github.com/benweissmann/memongo"
+	"github.com/cucumber/godog"
+	"github.com/gofrs/uuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"net/http"
+)
+
+type PermissionsComponent struct {
+	ErrorFeature   componenttest.ErrorFeature
+	svc            *service.Service
+	errorChan      chan error
+	MongoClient    *mongo.Mongo
+	Config         *config.Config
+	HTTPServer     *http.Server
+	ServiceRunning bool
+}
+
+func NewPermissionsComponent(mongoFeature *componenttest.MongoFeature) (*PermissionsComponent, error) {
+
+	f := &PermissionsComponent{
+		HTTPServer:     &http.Server{},
+		errorChan:      make(chan error),
+		ServiceRunning: false,
+	}
+
+	var err error
+
+	f.Config, err = config.Get()
+	if err != nil {
+		return nil, err
+	}
+
+	getMongoURI := fmt.Sprintf("localhost:%d", mongoFeature.Server.Port())
```

Comment:
Will this always be localhost? It not possibly consider making this part of the config too.

Reply:
Yes, I believe it will always be localhost as `memongo` spins up a mongodb instance locally. But it also provides a [URI()](https://github.com/benweissmann/memongo/blob/8e043283900b35ddae395e600fa77517af5d66bf/main.go#L155-L158) function which essentially does this so maybe we could use that instead:
```suggestion
	getMongoURI := mongoFeature.Server.URI()
```